src/game/client/components/menus_settings.cpp

```diff
@@ -1395,14 +1395,65 @@ void CMenus::RenderSettingsTee(CUIRect MainView)
 
 	// bottom button
 	float ButtonWidth = (BottomView.w/6.0f)-(SpacingW*5.0)/6.0f;
-	float BackgroundWidth = s_CustomSkinMenu||(m_pSelectedSkin && (m_pSelectedSkin->m_Flags&CSkins::SKINFLAG_STANDARD) == 0) ? ButtonWidth*2.0f+SpacingW : ButtonWidth;
+	float BackgroundWidth = ButtonWidth;
+	if (s_CustomSkinMenu)
+		BackgroundWidth = ButtonWidth*3.0f+SpacingW;
+	else if (m_pSelectedSkin && (m_pSelectedSkin->m_Flags&CSkins::SKINFLAG_STANDARD) == 0)
+		BackgroundWidth = ButtonWidth*2.0f+SpacingW;
 
 	BottomView.VSplitRight(BackgroundWidth, 0, &BottomView);
 	RenderBackgroundShadow(&BottomView, true);
 
 	BottomView.HSplitTop(25.0f, &BottomView, 0);
 	if(s_CustomSkinMenu)
 	{
+	BottomView.VSplitLeft(ButtonWidth, &Button, &BottomView);
+		static CButtonContainer s_RandomSkinButton;
+		if(DoButton_Menu(&s_RandomSkinButton, Localize("Random"), 0, &Button))
+		{
+			for(int p = 0; p < NUM_SKINPARTS; p++)
+			{
+				int Hue = random_int() % 255;
+				int Sat = random_int() % 255;
+				int Lgt = random_int() % 255;
+				int Alp = 0;
+				if (p == 1) // 1 == SKINPART_MARKING
+					Alp = random_int() % 255;
+				int NewVal = (Alp << 24) + (Hue << 16) + (Sat << 8) + Lgt;
+				*CSkins::ms_apColorVariables[p] = NewVal;
+			}
+			
+			static sorted_array<const CSkins::CSkinPart *> s_paList[6];
```

Comment:
This one shouldn't be needed. When it picks a special skin it could just pick another one.